src/parser.js

```diff
@@ -1638,6 +1642,26 @@ export class Parser extends Tokenizer {
     if (token instanceof IdentifierLikeToken) {
       return new Shift.StaticPropertyName(this.parseIdentifier().name);
     }
+    if (token.type == TokenType.LBRACK) {
+      let previousGeneratorParameter = this.paramGeneratorParameter;
+      let previousYield = this.paramYield;
+      this.expect(TokenType.LBRACK);
+      if (this.paramGeneratorParameter) {
+        // [+GeneratorParameter] ComputedPropertyName
+        this.paramGeneratorParameter = false;
+        this.paramYield = false;
+      } // else [~GeneratorParameter] ComputedPropertyName[?Yield]
+      let expr = this.parseAssignmentExpression();
+      token = this.lookahead;
+      if (token.type !== TokenType.RBRACK) {
```

Comment:
Won't expecting LBRAC change the error message from ErrorMessages.INVALID_PROPERTY_NAME to Expected '[', found ...